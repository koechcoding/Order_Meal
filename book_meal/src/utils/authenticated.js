export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}