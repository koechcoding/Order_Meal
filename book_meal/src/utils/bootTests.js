import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
