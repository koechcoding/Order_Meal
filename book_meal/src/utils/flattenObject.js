export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}