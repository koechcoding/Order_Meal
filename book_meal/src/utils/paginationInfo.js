export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}