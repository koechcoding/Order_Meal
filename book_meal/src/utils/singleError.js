const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;