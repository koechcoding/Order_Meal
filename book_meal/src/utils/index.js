import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}