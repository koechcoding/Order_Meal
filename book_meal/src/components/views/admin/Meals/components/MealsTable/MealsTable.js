import React from 'react';
import PropTypes from 'prop-types';
import Paginator from 'src/components/common/Paginator';
import { paginationInfo } from 'src/utils';

import Table from 'src/components/common/Table';
import { EntryType } from 'src/constants';

class MealsTable extends React.Component {

    render() {
        const { meals } = this.props.data;
        const tableData = {
            columns: [
                { key: 'id', title: 'ID', type: EntryType.NUMBER },
                { key: 'img_url', title: 'Image', type: EntryType.IMAGE },
                { key: 'name', title: 'Name', type: EntryType.TEXT },
                { key: 'cost', title: 'Cost', type: EntryType.NUMBER },
                { key: 'created_at', title: 'Created On', type: EntryType.DATE }
            ],
            rows: (meals) ? meals : []
        };
        const pageInfo = paginationInfo(this.props.data);
        const { toggleEdit, toggleDelete } = this.props;
        return (
            <div>
                <Table data={tableData} onEdit={toggleEdit} onDelete={toggleDelete} />
                <Paginator {...this.props} pageInfo={pageInfo} />
            </div>
        );
    }
}

MealsTable.propTypes = {
    data: PropTypes.object.isRequired,
}

export default MealsTable;