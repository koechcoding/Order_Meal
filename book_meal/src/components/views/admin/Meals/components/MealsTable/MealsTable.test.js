import React from 'react';
import { shallow } from 'enzyme';
import 'src/utils/bootTests';
import MealsTable from './MealsTable';
import Adapter from 'enzyme-adapter-react-16';

co