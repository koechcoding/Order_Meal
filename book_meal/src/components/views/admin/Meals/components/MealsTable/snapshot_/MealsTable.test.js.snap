// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealsTable /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MealsTable
    data={
        Object {
            "meals": Array [
              Object {
                "cost": 20,
                "created_at": "",
                "id": 1,
                "img_url": "#",
                "name": "ugali",
              },
            ],
          }
    }
    onNext={[Function]}
    onPrev={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Table
          data={
                    Object {
                              "columns": Array [
                                Object {
                                  "key": "id",
                                  "title": "ID",
                                  "type": 1,
                                },
                                Object {
                                  "key": "img_url",
                                  "title": "Image",
                                  "type": 2,
                                },
                                Object {
                                  "key": "name",
                                  "title": "Name",
                                  "type": 0,
                                },
                                Object {
                                  "key": "cost",
                                  "title": "Cost",
                                  "type": 1,
                                },
                                Object {
                                  "key": "created_at",
                                  "title": "Created On",
                                  "type": 3,
                                },
                              ],
                              "rows": Array [
                                Object {
                                  "cost": 20,
                                  "created_at": "",
                                  "id": 1,
                                  "img_url": "#",
                                  "name": "ugali",
                                },
                              ],
                            }
          }
          onDelete={undefined}
          onEdit={undefined}
/>,
        <Paginator
          data={
                    Object {
                              "meals": Array [
                                Object {
                                  "cost": 20,
                                  "created_at": "",
                                  "id": 1,
                                  "img_url": "#",
                                  "name": "ugali",
                                },
                              ],
                            }
          }
          onNext={[Function]}
          onPrev={[Function]}
          pageInfo={
                    Object {
                              "currentCount": undefined,
                              "currentPage": 1,
                              "hasNext": undefined,
                              "hasPrev": undefined,
                              "nextPage": undefined,
                              "pageCount": 1,
                              "perPage": undefined,
                              "prevPage": undefined,
                              "total": undefined,
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "columns": Array [
              Object {
                "key": "id",
                "title": "ID",
                "type": 1,
              },
              Object {
                "key": "img_url",
                "title": "Image",
                "type": 2,
              },
              Object {
                "key": "name",
                "title": "Name",
                "type": 0,
              },
              Object {
                "key": "cost",
                "title": "Cost",
                "type": 1,
              },
              Object {
                "key": "created_at",
                "title": "Created On",
                "type": 3,
              },
            ],
            "rows": Array [
              Object {
                "cost": 20,
                "created_at": "",
                "id": 1,
                "img_url": "#",
                "name": "ugali",
              },
            ],
          },
          "onDelete": undefined,
          "onEdit": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "meals": Array [
              Object {
                "cost": 20,
                "created_at": "",
                "id": 1,
                "img_url": "#",
                "name": "ugali",
              },
            ],
          },
          "onNext": [Function],
          "onPrev": [Function],
          "pageInfo": Object {
            "currentCount": undefined,
            "currentPage": 1,
            "hasNext": undefined,
            "hasPrev": undefined,
            "nextPage": undefined,
            "pageCount": 1,
            "perPage": undefined,
            "prevPage": undefined,
            "total": undefined,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Table
            data={
                        Object {
                                    "columns": Array [
                                      Object {
                                        "key": "id",
                                        "title": "ID",
                                        "type": 1,
                                      },
                                      Object {
                                        "key": "img_url",
                                        "title": "Image",
                                        "type": 2,
                                      },
                                      Object {
                                        "key": "name",
                                        "title": "Name",
                                        "type": 0,
                                      },
                                      Object {
                                        "key": "cost",
                                        "title": "Cost",
                                        "type": 1,
                                      },
                                      Object {
                                        "key": "created_at",
                                        "title": "Created On",
                                        "type": 3,
                                      },
                                    ],
                                    "rows": Array [
                                      Object {
                                        "cost": 20,
                                        "created_at": "",
                                        "id": 1,
                                        "img_url": "#",
                                        "name": "ugali",
                                      },
                                    ],
                                  }
            }
            onDelete={undefined}
            onEdit={undefined}
/>,
          <Paginator
            data={
                        Object {
                                    "meals": Array [
                                      Object {
                                        "cost": 20,
                                        "created_at": "",
                                        "id": 1,
                                        "img_url": "#",
                                        "name": "ugali",
                                      },
                                    ],
                                  }
            }
            onNext={[Function]}
            onPrev={[Function]}
            pageInfo={
                        Object {
                                    "currentCount": undefined,
                                    "currentPage": 1,
                                    "hasNext": undefined,
                                    "hasPrev": undefined,
                                    "nextPage": undefined,
                                    "pageCount": 1,
                                    "perPage": undefined,
                                    "prevPage": undefined,
                                    "total": undefined,
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "columns": Array [
                Object {
                  "key": "id",
                  "title": "ID",
                  "type": 1,
                },
                Object {
                  "key": "img_url",
                  "title": "Image",
                  "type": 2,
                },
                Object {
                  "key": "name",
                  "title": "Name",
                  "type": 0,
                },
                Object {
                  "key": "cost",
                  "title": "Cost",
                  "type": 1,
                },
                Object {
                  "key": "created_at",
                  "title": "Created On",
                  "type": 3,
                },
              ],
              "rows": Array [
                Object {
                  "cost": 20,
                  "created_at": "",
                  "id": 1,
                  "img_url": "#",
                  "name": "ugali",
                },
              ],
            },
            "onDelete": undefined,
            "onEdit": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "meals": Array [
                Object {
                  "cost": 20,
                  "created_at": "",
                  "id": 1,
                  "img_url": "#",
                  "name": "ugali",
                },
              ],
            },
            "onNext": [Function],
            "onPrev": [Function],
            "pageInfo": Object {
              "currentCount": undefined,
              "currentPage": 1,
              "hasNext": undefined,
              "hasPrev": undefined,
              "nextPage": undefined,
              "pageCount": 1,
              "perPage": undefined,
              "prevPage": undefined,
              "total": undefined,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;